### PR TITLE
Single invoice download

### PR DESCRIPTION
**Pull request by Shruti-Apte**
## Notion card
https://www.notion.so/saeloun/Add-Download-button-to-Invoices-List-page-API-Integration-eda269189b9e437ba2feada30448e86a
## Summary

- API integration for single invoice download
## Preview

https://www.loom.com/share/0fb454d34faf4048b64855f7f1e4f2fa

https://www.loom.com/share/533b48b07fc544d5b022023b31a24c12
## Type of change

- [ ] New feature (non-breaking change which adds functionality)

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -527,7 +527,6 @@ DEPENDENCIES
   activerecord-import
   annotate
   aws-sdk-s3
-  bcrypt (~> 3.1.7)
```
[the remark this answers is not in the document]
Comment (Shruti-Apte) [reply Gemfile.lock:531]: Fixed it!

File: app/javascript/src/components/Invoices/Invoice/index.tsx [hunk 2]
```diff
@@ -35,6 +36,20 @@ const Invoice = () => {
     }
   };
 
+  const handleDownloadInvoice = async () => {
+    try {
+      const response = await invoicesApi.downloadInvoice(invoice.id);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/index.tsx:41]: @Shruti-Apte can we name variable `response `to `res`

File: app/javascript/src/components/Invoices/common/MoreOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import { Trash, DownloadSimple } from "phosphor-react";
+
+const MoreOptions = ({ deleteInvoice, downloadInvoice }) => (
+  <ul className="py-2 absolute right-20 bg-white border-2 rounded border-miru-gray-200 drop-shadow">
+    {downloadInvoice != null && <li
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/common/MoreOptions.tsx:7]: @Shruti-Apte Boolean value of `null` is false. Can't we write `dowloadInvoice && `

File: app/javascript/src/components/Invoices/List/Table/TableRow.tsx [hunk 4]
```diff
@@ -39,6 +41,20 @@ const TableRow = ({
     }
   };
 
+  const handleDownload = async (id) => {
+    try {
+      const response = await invoicesApi.downloadInvoice(id);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/List/Table/TableRow.tsx:46]: @Shruti-Apte can we name the variable `response` to `res`. I think this `handleDownload` is getting used in different components. Can we re-use it? If not possible let's leave it like this.

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Invoices/common/MoreOptions.tsx (hunk 3 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Invoices/common/MoreOptions.tsx:7]: `dowloadInvoice` is a function. We are not using download functionality on edit invoice page (may use it in future), 
but that component is common that's why the condition `downloadInvoice != null &&`

**Comment by apoorv1316** [comment]
@Shruti-Apte We are getting the option to download the draft invoice on the invoice/id page.
Screenshot 2022-11-17 at 12 36 44 PM

**Review by prasanthchaduvula — APPROVED** [review]

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**